### DISCUSSION TITLE
Thank you for creating this tool! + feature request

### DISCUSSION THREAD

**Discussion by danfoerster** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#151] [Ideas]
First of all. Thank you for creating this little tool. I am running Big Sur 11.2.2 on a Late 2013 15" Retina MacBook Pro and it runs fine.

About two months ago I had to install a new battery into it to which I like to do my fair share to keep healthy for as long as I can. Since my MacBook is plugged in most of the time, and the battery health feature is not available on older machines, this is a great addition and I look forward to any new versions of it to see what else you guys can introduce, feature-wise. Although it does what it is supposed to already.

I miss a few features that would be great to have.
- A 'Fully charge now' button.
- Scheduled charging (So the computer is charging to 100% overnight if one requires a definite full charge the next morning)
- I also think the menu panel could be cosmetically refined a little bit

I do have my max percentage set to 81% which according to the built-in monitor is 85%. But like it is stated, it's only somewhat accurate. But looking at the actual percentage (from coconut battery) which is more accurate, the 81% of AlDente is basically spot on with coconut battery showing 80,8%.

<img width="352" alt="Screen Shot 2021-02-28 at 15 06 51" src="https://user-images.githubusercontent.com/79795921/109421197-a13ca300-79d6-11eb-86f3-90b17f227fb7.png">

The reason for this, and you surely know this, macOS and all other OS by apple use a very defensive monitoring. Meaning, the value the stock menu bar icon shows is never truly right. Just like, my Mac and iPhone and iPad will show 100% even though the percentage already is quite a bit lower. Usually at around 95% it will start to drop and then those two values gets closer and closer the lower the percentage drops. In order to make this accurate (as most users would probably prefer the AlDente value to be aligned with Apple's defensive value, you have to target the more inaccurate value, instead of the exact value (the actual percentage, for example coconut battery shows). I don't know if that is possible or even something that would make more sense.

I included some pictures. Maybe its a little inspiration to you. Keep up the great work. God bless for making a tool to do this on my 'old' machine. Kind regards from Austria.

![pic1](https://user-images.githubusercontent.com/79795921/109424245-347bd580-79e3-11eb-9916-3752c0e6deca.jpg)

![pic2](https://user-images.githubusercontent.com/79795921/109424247-35146c00-79e3-11eb-81ae-0589c4fbfc5e.jpg)

**Comment by danfoerster** [comment]
I also just realized that after quitting the app, the Mac does not proceed charging. Does this mean the set charge value is permanently written into the SMC (or as long as it isn't changed via AlDente) and doesn't 'reset' as long as the SMC is not reset? Without AlDente even running?

Also I just thought what would be nice as well, a 'Calibration' button. Which disables AlDente until a full charge and discharge cycle has been done and also maybe reminds people to do a calibration.

**Comment by davidwernhart** [comment]
Hey @danfoerster !
Thank you for your feedback. We are currently in the middle of developing the pro version and are definitly incorperating your suggestions.
Regarding your second question, with version 2.1 I built in a security feature into the helper, which automatically resets all modified SMC keys and assertions when the main application is closed. Give the OS a minute or two to update the charging state, but in theory your MacBook should charge as usual afterwards. The only exception is the classic key mode on intel machines, which is not reset on purpose for people who want the max charge value to persist restarts. You can reset this one by simply setting the charge to 100% in classic key mode.

Hope that helps,
David
